backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/ModularContractService.java

```diff
@@ -464,4 +459,63 @@ public void closeContractsOnOrderCloseIfNeededAndAllowed(@NonNull final I_C_Orde
 
 		modularAndInterimContracts.forEach(contractChangeBL::endContract);
 	}
+
+	private Set<I_C_Flatrate_Term> getModularAndInterimContracts(@NonNull final OrderId orderId)
+	{
+		return flatrateBL.getByOrderId(orderId).stream()
+				.filter(this::isModularOrInterimContract)
+				.collect(Collectors.toSet());
+	}
+
+	private Set<OrderId> getSalesOrderIdsLinkedToPurchaseModularContract(@NonNull final Set<FlatrateTermId> modularContractIds)
+	{
+		return orderBL.streamOrderLines(OrderLineQuery.builder().modularPurchaseContractIds(modularContractIds).build())
```

Comment:
hmm, did u just fall into the trap i was talking here: https://github.com/metasfresh/metasfresh/pull/20965/files#r2151787412 ?